source/mesh_deformation/interface.cc

```diff
@@ -386,7 +386,7 @@ namespace aspect
            p != x_no_flux_boundary_indicators.end(); ++p)
         if (mesh_deformation_boundary_indicators_set.find(*p) != mesh_deformation_boundary_indicators_set.end())
           {
-            x_no_flux_boundary_indicators.erase(*p);
+            p = x_no_flux_boundary_indicators.erase(p);
```

Comment:
That is not correct, because erase() returns the element after p, which you then increment ``++p``. This means you skip one entry. Instead you need to do
```
for (std::set<types::boundary_id>::const_iterator p = x_no_flux_boundary_indicators.begin();
           p != x_no_flux_boundary_indicators.end();)
if (bla)
   p = x_no_flux_boundary_indicators.erase(p);
else
  ++p;

```

Reply:
This is so hard to read (and remember) that I'd suggest using `std::set_difference` instead: https://en.cppreference.com/w/cpp/algorithm/set_difference

It would be nice to use `std::set::erase_if`, but that's only C++20 :-(

Reply:
Specifically, what you need here is
```
  std::set_difference (tangential_mesh_boundary_indicators.begin(),
                                 tangential_mesh_boundary_indicators.end(),
                                 mesh_deformation_boundary_indicators_set.begin(),
                                 mesh_deformation_boundary_indicators_set.end(),
                                 std::inserter (x_no_flux_boundary_indicators, x_no_flux_boundary_indicators.begin()));
```